File: RobotMicroservice/src/main/java/it/contrader/dto/BehaviorDTO.java
package it.contrader.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class BehaviorDTO {
	private Integer idBehavior;
	private String name;
	private ThingDTO thing = new ThingDTO();
}

File: AngularMicroservice/src/dto/BehaviorDTO.ts
import { ThingDTO } from './ThingDTO';

export class BehaviorDTO {
    idBehavior: number;
    name: String;
    thingDTO: ThingDTO;


constructor(id: number, name: String, idThing: number){
    this.idBehavior = id;
    this.name = name;
    idThing = this.thingDTO.id;
}
}